### PR TITLE
Improve sddp initialization

### PR DESCRIPTION
**Pull request by frapac**
- Initialize SDDP with random scenarios (instead computing scenarios with 0 as value functions)
- remove dead code in oneStepOneAlea
- fix non working cuts pruning

### CODE REVIEW

File: src/SDDPoptimize.jl [hunk 1]
```diff
@@ -323,10 +323,11 @@ function initialize_value_functions(model::SPModel,
...
                         param,
-                        solverProblems_null,
+                        solverProblems,
                         aleas,
                         true)[2]
 
+
     backward_pass!(model,
                   param,
                   V,
```
Comment (Henri-Gerard) [inline src/SDDPoptimize.jl:333]: Ok for this file

File: src/forwardBackwardIterations.jl [hunk 2]
```diff
@@ -73,27 +73,28 @@ function forward_simulations(model::SPModel,
...
+                    costs[k] += nextstep.cost - nextstep.cost_to_go
+                    if t==T-1
+                        costs[k] += nextstep.cost_to_go
+                    end
+                else
+                    stocks[t+1, k, :] = state_t
                 end
             else
-                stocks[t+1, k, :] = state_t
+                stocks[t+1, k, :] = [model.xlim[i][1] + rand()*(model.xlim[i][2] - model.xlim[i][1]) for i in 1:model.dimStates]
```
Comment (leclere) [inline src/forwardBackwardIterations.jl]: Two things : 
- commenting a bit the code could be quite useful (exchanging the if and else could help as well)
- I am not sure if we want to use forward_simulations to generate random state points. If the function is useful lets do it in another function, no ?

Comment (frapac) [inline src/forwardBackwardIterations.jl]: I agree to exchange the if with the else.

**Comment by frapac** [comment]
For the generation of random points, maybe we should let the user what he wants to do during the first forward pass. I see these options:
- init value functions as 0. and perform the first forward pass, then dump the first cut (that was implemented previously, but it is not the most efficient method I think)
- build first scenarios randomly (what this PR is about) 
- init value functions with V0_min, where V0_min is an estimation of the lower bound. As in Pfeiffer & al., we could estimate V0_min while solving the deterministic problem where random variables are approximated with their expectations. 

What do you think?

**Pull request closed without merging**